driver/http.go

```diff
@@ -550,6 +550,104 @@ func (g *httpStreamReader) close() error {
 	return g.closer.Close()
 }
 
+func (c *httpCRUD) search(ctx context.Context, collection string, req *SearchRequest) (SearchResultIterator, error) {
+	// Will anything break if we don't set transaction context here?
+	if req.SearchFields == nil {
+		req.SearchFields = []string{}
+	}
+	resp, err := c.api.TigrisSearch(ctx, c.db, collection, apiHTTP.TigrisSearchJSONRequestBody{
+		Q:            &req.Q,
+		SearchFields: &req.SearchFields,
+		Filter:       json.RawMessage(req.Filter),
+		Facet:        json.RawMessage(req.Facet),
+		Fields:       json.RawMessage(req.ReadFields),
+		Page:         &req.Page,
+		PageSize:     &req.PageSize,
+	})
+	err = HTTPError(err, resp)
+	dec := json.NewDecoder(resp.Body)
+
+	return &searchResultIterator{
+		searchStreamReader: &httpSearchReader{
+			closer: resp.Body,
+			stream: dec,
+		},
+		eof: err != nil,
+		err: err,
+	}, nil
+}
+
+type httpSearchReader struct {
+	closer io.Closer
+	stream *json.Decoder
+}
+
+func (g *httpSearchReader) read() (*SearchResponse, error) {
+	var res struct {
+		Result *apiHTTP.SearchResponse
+		Error  *api.ErrorDetails
+	}
+	if err := g.stream.Decode(&res); err != nil {
+		return nil, HTTPError(err, nil)
+	}
+
+	if res.Error != nil {
+		return nil, &Error{TigrisError: api.FromErrorDetails(res.Error)}
+	}
+
+	// TODO: is there a better way to implement conversion? This is error prone
```

Comment:
Need input - is there a better way to convert from `apiHTTP.SearchResponse` to `grpc.SearchResponse`? This conversion is error prone as there are a lot of pointer dereferencing.

Reply:
Yeah, we need to come up with something better.
At least for now we can avoid conversion of Facets and Meta by unmarshalling them directly into GRPC structs:
```
var res struct {
		Result struct {
            Hits   []*apiHTTP.SearchHit
		    Facets map[string]*api.SearchFacet
		    Meta   *api.SearchMetadata
        }
		Error  *api.ErrorDetails
	}
```

Reply:
thanks, addressed